common/app/views/fragments/image.scala.html

```diff
@@ -10,6 +10,19 @@
     isImmersiveMainMedia: Boolean = false
 )
 
+@immersivePortraitSource(largestImage: model.ImageAsset, hidpi: Boolean = false) = {
+    <source media="@if(hidpi) {
+        (orientation: portrait) and (-webkit-min-device-pixel-ratio: 1.25), (orientation: portrait) and (min-resolution: 120dpi)
+    } else {
+        (orientation: portrait)
+    }"
+    sizes="@{100 * (largestImage.width.toDouble / largestImage.height.toDouble)}vh"
+    srcset="@{(500 to 4000 by 250)
```

Comment:
Apart from https://github.com/guardian/frontend/pull/12727#issuecomment-217738761 (which we shouldn’t really care about) and the size of HTML, what are other downsides of using more finegrained steps? Cache?

I misunderstood. If `srcset="@{(500 to 4000 by 250)` means that there will always be 15 steps, the situation I was afraid of cannot occur. Does it mean we will be providing sources even over the specific image’s maximal resolution (`/master/[width].jpg` part of an URL)? That seems excessive, but may not be a problem if it’s only for immersive content when we target unknown viewport dimensions.


---NOT REALLY RELATED---
This reminds me of another tiny problem. Currently, if specific image’s maximal resolution is less than what could fully satisfy our `w=x&dpr=2&q=20` request, the image gets compressed at `q=20`. This is not a problem if its maximum resolution still satisfies what, say, would be `dpr=1,5`. But some of our imagery, wink, wink, nudge, nudge, do not even satisfy `dpr=1`! :laughing:

To pick on the usual offender, this comic native res is only 650px: http://www.theguardian.com/commentisfree/picture/2016/may/05/steve-bell-on-david-cameron-and-the-london-mayoral-campaign-cartoon

The above means that it will actually look worse on devices with `dpr`>1, because the image dimensions will be the same, but the `q` will be 20...

Not sure how to solve it, if only because the cutoff point would have to be discretionary. Shut upping now...